### ISSUE TITLE
make error, ldmd2 

### ISSUE THREAD

**Issue by ukreddy-erwin** [abraunegg/onedrive#533]
```
uday@jenkins-server:~/onedrive$ make clean; make;
rm -f onedrive onedrive.o version
rm -rf autom4te.cache
rm -f config.log config.status
if [ -f .git/HEAD ] ; then \
        echo v2.3.4-1-g56f6343 > version ; \
else \
        echo v2.3.5-dev > version ; \
fi
ldmd2  -w -g -O -J. -L-lcurl -L-lsqlite3  -L-ldl src/config.d src/itemdb.d src/log.d src/main.d src/monitor.d src/onedrive.d src/qxor.d src/selective.d src/sqlite.d src/sync.d src/upload.d src/util.d src/progress.d -ofonedrive
make: ldmd2: Command not found
Makefile:100: recipe for target 'onedrive' failed
make: *** [onedrive] Error 127
uday@jenkins-server:~/onedrive$
```

OS Details:
-----------
uday@jenkins-server:~/onedrive$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.2 LTS
Release:        18.04
Codename:       bionic

**Comment by abraunegg** [comment]
@ukreddy-erwin 
Please refer to the instructions regarding how to build correctly on Ubuntu. 

Based on your output above - you are not following the correct instructions: 

**Dependency Installation**
https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#dependencies-ubuntudebian---x86_64 

**Configure & Build**
https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#compilation--installation

**Comment by ukreddy-erwin** [comment]
I already tried these commands.

Please check my output:
sudo apt install build-essential

o/p:
Reading package lists... Done
Building dependency tree
Reading state information... Done
build-essential is already the newest version (12.4ubuntu1).
The following package was automatically installed and is no longer required:
  bridge-utils
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 135 not upgraded.

sudo apt install libcurl4-openssl-dev

o/p:
Reading package lists... Done
Building dependency tree
Reading state information... Done
libcurl4-openssl-dev is already the newest version (7.58.0-2ubuntu3.7).
The following package was automatically installed and is no longer required:
  bridge-utils
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 135 not upgraded.


sudo apt install libsqlite3-dev

o/p:
Reading package lists... Done
Building dependency tree
Reading state information... Done
libsqlite3-dev is already the newest version (3.22.0-1).
The following package was automatically installed and is no longer required:
  bridge-utils
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 135 not upgraded.


sudo apt install pkg-config

o/p:
Reading package lists... Done
Building dependency tree
Reading state information... Done
pkg-config is already the newest version (0.29.1-0ubuntu2).
The following package was automatically installed and is no longer required:
  bridge-utils
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 135 not upgraded.


curl -fsS https://dlang.org/install.sh | bash -s dmd

o/p:
dmd-2.086.0 already installed
Using dub 1.15.0 shipped with dmd-2.086.0

Run `source ~/dlang/dmd-2.086.0/activate` in your shell to use dmd-2.086.0.
This will setup PATH, LIBRARY_PATH, LD_LIBRARY_PATH, DMD, DC, and PS1.
Run `deactivate` later on to restore your environment.

sudo apt install libnotify-dev
o/p:
Reading package lists... Done
Building dependency tree
Reading state information... Done
libnotify-dev is already the newest version (0.7.7-3).
The following package was automatically installed and is no longer required:
  bridge-utils
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 135 not upgraded.

git clone https://github.com/abraunegg/onedrive.git
cd onedrive
./configure
make clean; make;
sudo make install

./configure
o/p:
checking for a BSD-compatible install... /usr/bin/install -c
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for dmd... no
checking for ldmd2... no
checking for ldc2... no
configure: error: Could not find any compatible D compiler

uday@jenkins-server:~/onedrive$ make clean; make;
rm -f onedrive onedrive.o version
rm -rf autom4te.cache
rm -f config.log config.status
if [ -f .git/HEAD ] ; then \
        echo v2.3.4-1-g56f6343 > version ; \
else \
        echo v2.3.5-dev > version ; \
fi
ldmd2  -w -g -O -J. -L-lcurl -L-lsqlite3  -L-ldl src/config.d src/itemdb.d src/log.d src/main.d src/monitor.d src/onedrive.d src/qxor.d src/selective.d src/sqlite.d src/sync.d src/upload.d src/util.d src/progress.d -ofonedrive
make: ldmd2: Command not found
Makefile:100: recipe for target 'onedrive' failed
make: *** [onedrive] Error 127

**Comment by abraunegg** [comment]
@ukreddy-erwin 
Please can you read very carefully your output and the links provided. Your output indicates what is wrong and you are missing 1 crucial step.

**Comment by ukreddy-erwin** [comment]
Configure: error: Could not find any compatible D compiler
But I already installed the dependencies you specified.

Can you please guide me, which step is missing.

**Comment by ukreddy-erwin** [comment]
I activated the compiler..
`source ~/dlang/dmd-2.086.0/activate` 

But still same output.

**Comment by abraunegg** [comment]
You need to:
1. Activate dmd
2. Run make clean
3. Run ./configure
4. Run make

**Comment by ukreddy-erwin** [comment]
uday@jenkins-server:~/onedrive$ `source ~/dlang/dmd-2.086.0/activate`
uday@jenkins-server:~/onedrive$ make clean
rm -f onedrive onedrive.o version
rm -rf autom4te.cache
rm -f config.log config.status
uday@jenkins-server:~/onedrive$ ./configure
checking for a BSD-compatible install... /usr/bin/install -c
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for dmd... no
checking for ldmd2... no
checking for ldc2... no
configure: error: Could not find any compatible D compiler


THese are steps I did.

**Comment by ukreddy-erwin** [comment]
uday@jenkins-server:~/onedrive$ ls ~/dlang/dmd-2.086.0
activate       html         linux  README.TXT  src
activate.fish  license.txt  man    samples

**Comment by abraunegg** [comment]
I suggest you totally start from scratch.

Uninstall all your Ubuntu packages for dmd / ldc and whatever else. Dmd is not activating for you correctly - I cant assist with that. When it activates correctly, it will prepend your shell prompt that it is activated.

Also - please for any future post, please use correct markdown language to escape your cli output to ensure it is more readable

**Comment by ukreddy-erwin** [comment]
`source ~/dlang/dmd-2.086.0/activate`

This is not activating the compiler as you told.
So I tried below steps.

mkdir -p ~/dlang && wget https://dlang.org/install.sh -O ~/dlang/install.sh
~/dlang/install.sh dmd -a

This is activating the compiler and i am able to make the onedrive.

**Comment by abraunegg** [comment]
> `source ~/dlang/dmd-2.086.0/activate`
> 
> This is not activating the compiler as you told.
> So I tried below steps.
> 
> mkdir -p ~/dlang && wget https://dlang.org/install.sh -O ~/dlang/install.sh
> ~/dlang/install.sh dmd -a
> 
> This is activating the compiler and i am able to make the onedrive.

Thanks for the feedback - however this suggest something VERY environmental to your Ubuntu system, as the instructions to install DMD are as per https://dlang.org/download.html and utilising curl:

```
curl -fsS https://dlang.org/install.sh | bash -s dmd
```

This is the same instruction as listed under here: https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#build-requirements

If that is not working for you, then there is some sort of localised issue on your system that is preventing that from working.

**Comment by norbusan** [comment]
@ukreddy-erwin what is the shell you are running? bash?

@abraunegg could it be that the `source` does not work in some shells? Or differently?

**Comment by abraunegg** [comment]
@norbusan 
No idea - however if it is a stock standard `18.04.2 LTS` system - then it should work without issue - I tested it, plus teh instructions on the dlang site would also be different. There is certainly something wrong with the local operating environment to cause this.

Depending on how DMD was installed in the first place:
>curl -fsS https://dlang.org/install.sh | bash -s dmd
>
>o/p:
>dmd-2.086.0 already installed
>Using dub 1.15.0 shipped with dmd-2.086.0


This 'kind of' indicates to me that DMD was potentially initially installed via packages - which are not entirely working on Ubuntu ...

**Comment by norbusan** [comment]
Ahh, that could indeed be ...
